pandas/tests/indexes/interval/test_interval.py

```diff
@@ -400,28 +400,72 @@ def test_get_item(self, closed):
                                              closed=closed)
         tm.assert_index_equal(result, expected)
 
-    # To be removed, replaced by test_interval_new.py (see #16316, #16386)
-    def test_get_loc_value(self):
-        pytest.raises(KeyError, self.index.get_loc, 0)
-        assert self.index.get_loc(0.5) == 0
-        assert self.index.get_loc(1) == 0
-        assert self.index.get_loc(1.5) == 1
-        assert self.index.get_loc(2) == 1
-        pytest.raises(KeyError, self.index.get_loc, -1)
-        pytest.raises(KeyError, self.index.get_loc, 3)
-
-        idx = IntervalIndex.from_tuples([(0, 2), (1, 3)])
-        assert idx.get_loc(0.5) == 0
-        assert idx.get_loc(1) == 0
-        tm.assert_numpy_array_equal(idx.get_loc(1.5),
-                                    np.array([0, 1], dtype='int64'))
-        tm.assert_numpy_array_equal(np.sort(idx.get_loc(2)),
-                                    np.array([0, 1], dtype='int64'))
-        assert idx.get_loc(3) == 1
-        pytest.raises(KeyError, idx.get_loc, 3.5)
-
-        idx = IntervalIndex.from_arrays([0, 2], [1, 3])
-        pytest.raises(KeyError, idx.get_loc, 1.5)
+    def check_get_loc_result(self, idx, key, correct):
+        """
+        helper function to check the result for get_loc related tests
+
+        Parameters
+        ----------
+
+        idx: IntervalIndex
+            IntervalIndex over which get_loc checks will be run
+
+        key: scalar or Interval
+
+        correct: dict
```

Comment:
not a fan of this. quickly these test routines get very bespoke, and are not well known and hence repeated often enough. This is doing too much; should not catch raises. Its *ok* that it can compare an array and scalar though.

Maybe if you make a general one thats a function (and not restricted to comparisions here) would be ok.